src/widgets/buttons.ts

```diff
@@ -478,6 +488,42 @@ export function deleteButtonWithCheck (
   return deleteButtonElt
 }
 
+function getButtonStyle (options: ButtonWidgetOptions = { buttonType: ButtonType.Primary, filled: true }) {
+  // We need to accomadate for legacy code, which is why we have to allow buttonType and filled to be optional
+  let backgroundColor: (ButtonType | string | undefined) = options.buttonType
```

Comment:
It's confusing to have two variables that mean the same thing but one is called type and the other is called color. Also having a variable of type `(ButtonType | string | undefined)` feels like it's an intermediate result; the end result you're working towards is a string, right? So maybe just use an if statement here, so:
```ts
let backgroundColor: string = PRIMARY_COLOR;
if (options.buttonType === Secondary) {
  backgroundColor = SECONDARY_COLOR;
}
```

Reply:
I see what you are saying. I'll look a little further as I really thought it was better to use an enum.  I had to do undefined because of the legacy code, some may not pass in an option and it may be undefined.. basically typescript complained.

Reply:
See https://github.com/solid/solid-ui/pull/289#discussion_r400060297 and https://github.com/solid/solid-ui/pull/289#discussion_r400066550